src/function/table/table_scan.cpp

```diff
@@ -402,10 +388,31 @@ unique_ptr<vector<unique_ptr<Expression>>> ExtractFilters(const ColumnDefinition
                                                           idx_t storage_idx) {
 	ColumnBinding binding(0, storage_idx);
 	auto bound_ref = make_uniq<BoundColumnRefExpression>(col.Name(), col.Type(), binding);
-
 	auto filter_expressions = make_uniq<vector<unique_ptr<Expression>>>();
-	ExtractInFilter(filter, *bound_ref, filter_expressions);
 
+	switch (filter->filter_type) {
+	case TableFilterType::OPTIONAL_FILTER:
+		// Conventional IN filter.
+		ExtractOptionalFilter(*filter, *bound_ref, filter_expressions, false);
```

Comment:
Currently all `IN` filters are optional - but that's not necessarily required - can we just go into the optional filter and not treat it special here